expression/expression.go

```diff
@@ -996,13 +996,22 @@ func scalarExprSupportedByTiKV(sf *ScalarFunction) bool {
 
 func scalarExprSupportedByFlash(function *ScalarFunction) bool {
 	switch function.FuncName.L {
+	case ast.Floor, ast.Ceil, ast.Ceiling:
+		switch function.Function.PbCode() {
+		case tipb.ScalarFuncSig_FloorIntToDec, tipb.ScalarFuncSig_CeilIntToDec:
+			return false
+		default:
+			return true
+		}
 	case
 		ast.LogicOr, ast.LogicAnd, ast.UnaryNot, ast.BitNeg, ast.Xor, ast.And, ast.Or,
 		ast.GE, ast.LE, ast.EQ, ast.NE, ast.LT, ast.GT, ast.In, ast.IsNull, ast.Like,
 		ast.Plus, ast.Minus, ast.Div, ast.Mul, /*ast.Mod,*/
 		ast.If, ast.Ifnull, ast.Case,
-		ast.Month,
-		ast.TimestampDiff, ast.DateFormat, ast.FromUnixTime,
+		ast.Concat, ast.ConcatWS,
```

Comment:
`Concat`, `ConcatWS`, `Year`, `Month`, `Day`, `DateDiff` should be introduced by https://github.com/pingcap/tidb/pull/25145.

Rest LGTM.

Reply:
https://github.com/pingcap/tidb/pull/25085/files
Concat, ConcatWS, Year, Month, Day, DateDiff are already there in this PR
Now they are needed to be discarded, right?

Reply:
> https://github.com/pingcap/tidb/pull/25085/files
> Concat, ConcatWS, Year, Month, Day, DateDiff are already there in this PR
> Now they are needed to be discarded, right?

Yes, but they are not in release-5.0 yet...

Reply:
done